### DISCUSSION TITLE
How to test in a multi-tenant scenario

### DISCUSSION THREAD

**Discussion by willwainwright** [microsoft/Microsoft-Fabric-workload-development-sample#111] [Help] [answered]
We want to test our workload externally by installing our nupkg file in a test customer tenant.
However during workload upload we get the error: "ResourceId domain - XXX is not in the tenant domains list".

Is there a way we can install this on test customer tenants before going through the publishing workflow?

**Comment by gregahren** [comment]
+1

**Comment by ramizsha** [comment, marked as answer]
Please refer to this https://learn.microsoft.com/en-us/fabric/workload-development-kit/publish-workload-flow#process .
After that you will be able to publish it to up to 10 test tenants (see https://learn.microsoft.com/en-us/fabric/workload-development-kit/publish-workload-flow#preview-audience)
